source/Octopus.Tentacle/Commands/DeregisterMachineCommand.cs

```diff
@@ -48,22 +51,30 @@ public class DeregisterMachineCommand : AbstractStandardCommand
 
         protected override void Start()
         {
+            cancellationTokenSource = new CancellationTokenSource();
             base.Start();
-            StartAsync().GetAwaiter().GetResult();
+            StartAsync(cancellationTokenSource.Token).GetAwaiter().GetResult();
+        }
+
+        protected override void Stop()
+        {
+            base.Stop();
+
+            cancellationTokenSource?.Cancel();
```

Comment:
I'm not sure about this - If I'm using `GetAwaiter().GetResult();` does it make much sense to create a token and cancel it on `Stop()`? As the `Start()` method won't complete until `StartAsync()` has completed.

Reply:
Hrm, it depends on how `Stop()` get's called. Does it get called in the same method chain as the `Start()`?

What is the purpose of `Stop()`? Is it to stop a long-running command?

Reply:
Start and Stop are dictated by the `ICommandHost` being used. 

The implementations are:
<img width="350" alt="Screenshot 2023-10-06 at 10 04 38" src="https://github.com/OctopusDeploy/OctopusTentacle/assets/97430840/1f969ca5-7175-42c5-87ee-cd9936adfc14">

I guess if we wanted to keep the existing functionality, we could just pass in `CancellationToken.None`.

Reply:
The stop should be getting called from a different thread that is listening to termination signals. 
In theory the `Start` method is blocking in both scenarios, really the `Start()` method itself should be async as well all the way up to the main entrypoint.

The task cancellation (e.g. `Ctrl+C`) should be signaled on its own thread and therefore setting this cancellation should  provide the signal to all the later tasks using that cancellation token. Ultimately the process will still be killed, but the signal provides an opportunity for a "graceful shutdown".

Reply:
Since it probably wasn't graceful before its probably not much different to _not_ provide the cancellation token, but then that loses some of the nice benefits of the async+cancellation token capabilities.